Core/src/org/sleuthkit/autopsy/centralrepository/datamodel/CorrelationAttributeNormalizer.java

```diff
@@ -0,0 +1,164 @@
+     * 
+     * @return normalized data
+     */
+    public static String normalize(CorrelationAttribute.Type attributeType, String data) throws CorrelationAttributeNormalizationException {
+        
+        final String errorMessage = "Validator function not found for attribute type: %s";
+        
+        if(attributeType == null){
+            throw new CorrelationAttributeNormalizationException(String.format(errorMessage, "null"));
+        }
+        
+        switch(attributeType.getId()){
+            case CorrelationAttribute.FILES_TYPE_ID:
+                return normalizeMd5(data);
+            case CorrelationAttribute.DOMAIN_TYPE_ID:
+                return normalizeDomain(data);
+            case CorrelationAttribute.EMAIL_TYPE_ID:
+                return normalizeEmail(data);
+            case CorrelationAttribute.PHONE_TYPE_ID:
+                return normalizePhone(data);
+            case CorrelationAttribute.USBID_TYPE_ID:
+                return normalizeUsbId(data);
+            default:
+                throw new CorrelationAttributeNormalizationException(String.format(errorMessage, attributeType.getDisplayName()));   
+        }
+    }
+    
+    /**
+     * Validate the data.  Converts text to lower case, and ensures that the
+     * data is a valid string of the format expected given the attributeType.
+     *  
+     * @param attributeTypeId correlation type of data
+     * @param data data to normalize
+     * 
+     * @return normalized data
+     */
+    public static String normalize(int attributeTypeId, String data) throws CorrelationAttributeNormalizationException {
+        try {
+            List<CorrelationAttribute.Type> defaultTypes = CorrelationAttribute.getDefaultCorrelationTypes();
+            Optional<CorrelationAttribute.Type> typeOption = defaultTypes.stream().filter(attributeType -> attributeType.getId() == attributeTypeId).findAny();
+            
+            if(typeOption.isPresent()){
+                CorrelationAttribute.Type type = typeOption.get();
+                return CorrelationAttributeNormalizer.normalize(type, data);
+            } else {
+                throw new CorrelationAttributeNormalizationException(String.format("Given attributeTypeId did not correspond to any known Attribute: %s", attributeTypeId));
+            }
+        } catch (EamDbException ex) {
+            throw new CorrelationAttributeNormalizationException(ex);
+        }
+    }
+
+    /**
+     * Verify MD5 is the correct length and values. Make lower case.
+     */
+    private static String normalizeMd5(String data) throws CorrelationAttributeNormalizationException {
+        final String errorMessage = "Data purporting to be an MD5 was found not to comform to expected format: %s";
+        if(data == null){
+            throw new CorrelationAttributeNormalizationException(String.format(errorMessage, data));
+        }
+        final String validMd5Regex = "^[a-fA-F0-9]{32}$";
```

Comment:
There's no point in matching the upper case hex digits since you convert the string to lower case before applying the regex.